core/src/main/java/org/infinispan/expiration/ExpirationManager.java

```diff
@@ -29,7 +30,17 @@
    void processExpiration();
 
    /**
-    * @return true if expiration is enabled, false otherwise
+    * @return true if expiration reaper is enabled, false otherwise
     */
    boolean isEnabled();
+
+   /**
+    * This should be invoked passing in an entry that is now expired.  Note this method is assumed to not have the
```

Comment:
The lock here is not the lock we keep but rather the lock in the underlying data container.  I will go back and triple check this is sufficient for the store case.